Fixed critical issues

NullSoldier opened the pull request:
I wanted to use your library but when I pulled your code down it didn't even compile. On top of that, reporting page views didn't work at all. I also noticed you removed events from your example so I re-added it back in.

Another major change is that I've removed the dependency of the app.config file. If you use TrackMe, it will still look in your app.config file for the account code; but otherwise if you invoke the library directly (which is what I wanted) then you can now specify the Google account code.

Critical fixes:
- Page view requests were not sent correctly causing them to never be reported
- The project did not compile because of the sample

Improvements:
- Removed dependency on app.config file
- Added derived files to .gitignore so they won't be added to the repository
- Removed lots of unused code, there's no reason for it. You can get it from your GIT history if you need it again
- Since the data objects (GooglePageView and GoogleEvent) are immutable (and should be), there's no need for validation functions. I moved validation to the constructor.
- Lots of XML comments referenced non-existent parameters. I removed them

In short, I fixed lots of things. The example now works, the code now works. I've tested it.

NullSoldier committed:
Added git ignore for derived files

Ignored bin, obj folders and the SUO

NullSoldier committed:
Added a 2012 SLN and upgraded the .csproj

NullSoldier committed:
Removed unused using directives

NullSoldier committed:
Fixed broken demo namespace reference

NullSoldier committed:
Cleaned up demo config

NullSoldier committed:
Cleaned up PageView

NullSoldier committed:
Cleaned up GoogleTracking

- Removed Xml docs that referred to parameters which no longer exist
- Removed tons of extra spacing
- Removed an unused class

NullSoldier committed:
Added Domain to GoogleEvent

This is to keep it consistent with the API for PageView. I also removed
Validate and did validation in the constructor.

NullSoldier committed:
Minor code cleanup to sample

NullSoldier committed:
Removed/changed comments in TrackMe

NullSoldier committed:
Moved Transaction validation into constructor

NullSoldier committed:
Added google event tracking back to example

NullSoldier committed:
Removed invalidate from tracker

NullSoldier committed:
Converted factory creating to object initializers

NullSoldier committed:
Fixed crash with no page title

Page title should not be required as GoogleEvents don't need a page title

NullSoldier committed:
Removed invalidate from TrackingRequest

NullSoldier committed:
Fixed TrackingRequest doing page view requests

Page view request were broken because it tried to submit "page" as the
request type when it did a page view. For some reason, when you submit
"page" the page view never gets counted. It disapears into the neather.
However, you don't even need to include it at all because if you don't
specify a type then the documentation says it will be a page view.

- Removed a ton of duplicated code using three helper functions
- Fixed TrackingRequest as a page view
- Changed properties to functions (they should be!)
- Removed unused code

NullSoldier committed:
Added .user file  to .gitignore

NullSoldier committed:
Removed app.config file dependency

Now the google analytics ID is passed in to the Factory. Optionally, if
you do use the app.config file you can just use
ConfigurationSettings.GoogleAccountCode to get it from your app config
without writing any new code. The only difference is that it does not do
this automatically now.

The reason for this is that I wanted to use this library in a plugin where
I do not wish to have app.config file dependencies.

NullSoldier committed:
Minor spacing changes

NullSoldier committed:
Dropped targeted framework to .NET 2.0

NullSoldier committed:
Fixed serializing page titles with spaces

NullSoldier committed:
Fixed event request being malformed

NullSoldier commented:
I also found two more bugs in my refactoring. As soon as you review/approve this pull request I would like to submit another one with fixes in TrackRequest for events, and page titles with spaces in the name.

dougrathbone commented:
hi there,

appreciate your input. I'm a bit snowed under, but will be able to review this this weekend. If nothing was working, was it related to the version of the analytic api in use (the last pull request changed it)...?

NullSoldier commented:
Unfortunately, I did not git blame to find out when it was broken. I'm
guessing it was written for an older version of the API and the last person
updated it. Either way, mine fixes it.

I did notice that there were a few more bugs that I fixed as well. (I broke
events in my pull request). I also changed the API so it can easily work in
a non ASP.net project without relying on an app config file. It shouldn't
break the API for anyone using the TrackMe. (It'll still grab it from the
app config in that case).

I'll submit another pull request to fix breaking events and another bug
after you review this one.

On Mon, Apr 22, 2013 at 8:09 PM, Doug Rathbone notifications@github.comwrote:

> hi there,
> 
> appreciate your input. I'm a bit snowed under, but will be able to review
> this this weekend. If nothing was working, was it related to the version of
> the analytic api in use (the last pull request changed it)...?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/dougrathbone/Ga.net/pull/5#issuecomment-16836804
> .
